nsfc-plugin/src/main/java/org/osc/controller/nsfc/api/NeutronSfcSdnRedirectionApi.java

```diff
@@ -147,45 +380,62 @@ public NetworkElement registerNetworkElement(List<NetworkElement> portPairGroupL
-                                        portPairGroupList.get(0).getParentId());
+        return retVal;
     }
 
     @Override
     public NetworkElement updateNetworkElement(NetworkElement serviceFunctionChain, List<NetworkElement> portPairGroupList)
             throws Exception {
-        this.utils.throwExceptionIfNullElementAndId(serviceFunctionChain, "Port Pair Group Service Function Chain Id");
-        this.utils.throwExceptionIfNullOrEmptyNetworkElementList(portPairGroupList, "Port Pair Group update member list");
+        checkArgument(serviceFunctionChain != null && serviceFunctionChain.getElementId() != null,
+                "null passed for %s !", "Service Function Chain Id");
+        throwExceptionIfNullOrEmptyNetworkElementList(portPairGroupList, "Port Pair Group update member list");
 
-        PortChain portChain = this.osClient.sfc().portchains().get(serviceFunctionChain.getElementId());
-        this.utils.throwExceptionIfCannotFindById(portChain, "Service Function Chain", serviceFunctionChain.getElementId());
+        PortChain portChain = this.osCalls.getPortChain(serviceFunctionChain.getElementId());
+        checkArgument(portChain != null && portChain.getId() != null,
+                      "Cannot find %s by id: %s!", "Service Function Chain", serviceFunctionChain.getElementId());
 
         portChain = Builders.portChain().from(portChain)
                             .portPairGroups(Collections.emptyList()).build();
-        this.osClient.sfc().portchains().update(portChain.getId(), portChain);
+        this.osCalls.updatePortChain(portChain.getId(), portChain);
         this.utils.validatePPGList(portPairGroupList);
 
         List<String> portPairGroupIds = portPairGroupList
                 .stream()
                 .map(ppg -> ppg.getElementId())
-                .collect(Collectors.toList());
+                .collect(toList());
 
-        portChain = Builders.portChain().from(portChain).portPairGroups(portPairGroupIds).build();
-        PortChain portChainUpdated = this.osClient.sfc().portchains().update(portChain.getId(), portChain);
-        return new NetworkElementEntity(portChainUpdated.getId(),
-                                        new ArrayList<>(), new ArrayList<>(),
-                                        portPairGroupList.get(0).getParentId());
+        portChain = Builders.portChain().portPairGroups(portPairGroupIds).build();
+        PortChain portChainUpdated = this.osCalls.updatePortChain(serviceFunctionChain.getElementId(), portChain);
+
+        List<PortPairGroupEntity> portPairGroups =
+                portPairGroupIds.stream().map(id -> new PortPairGroupEntity(id)).collect(toList());
+        ServiceFunctionChainEntity retVal = new ServiceFunctionChainEntity(portChainUpdated.getId());
+        portPairGroups.stream().forEach(p -> p.setServiceFunctionChain(retVal));
+        retVal.setPortPairGroups(portPairGroups);
+        return retVal;
```

Comment:
retvalue is just a Network Element. setting port pair groups on this does not make sense as the caller will never be able to see this value unless he does an unsafe cast